examples/stargate-jwt-auth/README.md

```diff
@@ -0,0 +1,43 @@
+This directory includes an example to start a keycloak server along with a Cassandra cluster and a Stargate node.
+
+# Run demo
+
+Use the `start_jwt.sh` shell script to run `docker-compose.yml` to start all the services, along with a `docker run` command
+to insert some CQL data for use with the example.
+```bash
+./start_jwt.sh
+```
+
+The `stargate-realm.json` file is used by the keycloak service. 
+
+The CQL files in the `scripts` directory are mounted and run using a docker-cqlsh image.
+The `cqlshrc` file gives the docker-cqlsh image user permissions to insert the data.
+
+# Additional information on what the CQL script does:
+
+## Create some data
+
+### Create a role in Cassandra
+
+```cql
+CREATE ROLE IF NOT EXISTS 'web_user' WITH PASSWORD = 'web_user' AND LOGIN = TRUE;
+```
+
+### Insert some data into Cassandra
+
+```cql
+CREATE KEYSPACE IF NOT EXISTS store WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':'1'};
+
+CREATE TABLE IF NOT EXISTS store.shopping_cart (userid text PRIMARY KEY, item_count int, last_update_timestamp timestamp);
+
+INSERT INTO store.shopping_cart (userid, item_count, last_update_timestamp) VALUES ('9876', 2, toTimeStamp(toDate(now())));
+INSERT INTO store.shopping_cart (userid, item_count, last_update_timestamp) VALUES ('1234', 5, toTimeStamp(toDate(now())));
+```
+
+### Grant role permissions to the table
+
+```cql
+CREATE ROLE IF NOT EXISTS 'web_user' WITH PASSWORD = 'web_user' AND LOGIN = TRUE;
+GRANT MODIFY ON TABLE store.shopping_cart TO web_user;
+GRANT SELECT ON TABLE store.shopping_cart TO web_user;
+```
```

Comment:
Not sure where you want to put it but this is a smoke test for JWT auth via REST

```
USER_TOKEN=$(curl -s --data "username=testuser1&password=testuser1&grant_type=password&client_id=user-service" http://localhost:4444/auth/realms/stargate/protocol/openid-connect/token | jq -r '.access_token')


curl -sL 'localhost:8082/v1/keyspaces/system/tables/local/rows/local' \
-H "X-Cassandra-Token: $USER_TOKEN" | jq .


curl -sL 'localhost:8082/v1/keyspaces/store/tables/shopping_cart/rows/9876' \
-H "X-Cassandra-Token: $USER_TOKEN" | jq .
{
  "count": 1,
  "rows": [
    {
      "item_count": 2,
      "userid": "9876",
      "last_update_timestamp": "2020-11-06T00:00:00Z"
    }
  ]
}

curl -sL 'localhost:8082/v1/keyspaces/store/tables/shopping_cart/rows/1234' \
-H "X-Cassandra-Token: $USER_TOKEN" | jq .
{
  "description": "Role unauthorized for operation: Not allowed to access this resource",
  "code": 401
}
```

Reply:
These tests are shown in the docs. I guess I can either add as a script in this repo, or just add a link to the docs in the README.